UniswapDataApi/Functions/GetUniswapOrderbook.cs

```diff
@@ -0,0 +1,153 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using UniswapDataApi.Models;
+
+namespace UniswapDataApi.Functions
+{
+    public class GetUniswapOrderBook
+    {
+        private readonly HttpClient _client;
+        private readonly int _cacheExpirationSecs;
+        private readonly string _getUniswapDataUri;
+
+        private const string DecimalFormatter = "0.##########";
+
+        private static DateTime _lastUpdated = DateTime.MinValue;
+        private static IDictionary<string, OrderBook> _orderBooks = new Dictionary<string, OrderBook>();
+
+        public GetUniswapOrderBook(IHttpClientFactory httpClientFactory, IConfiguration config)
+        {
+            _client = httpClientFactory.CreateClient();
+            _cacheExpirationSecs = int.TryParse(config["CacheExpirationSecs"], out var cacheExpSecs)
+                ? cacheExpSecs
+                : 30;
+            _getUniswapDataUri = config["GetUniswapDataUri"];
+            if (string.IsNullOrEmpty(_getUniswapDataUri))
+                throw new ApplicationException("'GetUniswapDataUri' must be defined in your configuration");
+        }
+
+        [FunctionName("orderBook")]
+        public async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route= "orderBook/{tokenSymbol}")] HttpRequest req,
+            string tokenSymbol,
+            ILogger log)
+        {
+            try
+            {
+                if (OrderBooksAreStale())
+                    await UpdateOrderBooks();
+
+                _orderBooks.TryGetValue(tokenSymbol.ToLower(), out var orderBook);
+                return new OkObjectResult(orderBook);
+            }
+            catch (Exception e)
+            {
+                log.LogCritical(e, $"Caught an exception while processing blocklytics <=> uniswap summary data: {e.Message}");
+                return new InternalServerErrorResult();
+            }
+        }
+
+        private async Task UpdateOrderBooks()
+        {
+            var response = await _client.GetAsync(_getUniswapDataUri);
+            var dtoString = await response.Content.ReadAsStringAsync();
+            var pairs = JsonConvert.DeserializeObject<IEnumerable<UniswapPair>>(dtoString);
+
+            var orderBooks = new Dictionary<string, OrderBook>();
+            foreach (var pair in pairs)
+            {
+                var tokenSymbol = pair.TokenSymbol();
+                if (orderBooks.ContainsKey(tokenSymbol))
+                    continue;
+
+                orderBooks.Add(tokenSymbol, PairToOrderBook(pair));
+            }
+
+            _orderBooks = orderBooks;
+            _lastUpdated = DateTime.UtcNow;
+        }
+
+        private static OrderBook PairToOrderBook(UniswapPair pair)
+        {
+            var ethLiquidity = double.Parse(pair.EthLiquidity);
```

Comment:
I'm not familiar with the Uniswap API - does this really come in as a decimal/double value? I would think it would be a BigInteger

Reply:
If it's not coming back as an integer, use the decimal type - it has higher precision and ETH in particular has 18 decimal places.

Reply:
Planning to convert to BigInteger or decimal. This is the API that's being used btw:
https://docs.blocklytics.org/apis/uniswap-api#exchanges

Reply:
Uploaded json response here if you're interested:
https://filebin.net/09r310uur18iiyb1/UniswapExchange.json?t=ui7yjvbl

Appears to go up to 10 decimal places